src/Actions/Media/ReplaceMedia.php

```diff
@@ -116,17 +125,11 @@ protected function validateData(): void
         $this->data['name'] = $this->data['name'] ?? $this->data['file_name'];
         $this->data['collection_name'] ??= 'default';
 
```

Comment:
validate if $this->data['collection_name'] even exists for given model. And check for readonly.